### ISSUE TITLE
[java] False negative in ClassWithOnlyPrivateConstructorsShouldBeFinal 

### ISSUE THREAD

**Issue by Chiky777** [pmd/pmd#5072]
**Affects PMD Version: 7.2.0** 

**Rule: ClassWithOnlyPrivateConstructorsShouldBeFinal**
<https://docs.pmd-code.org/latest/pmd_rules_java_design.html#classwithonlyprivateconstructorsshouldbefinal>

**Description:**
I found a regresson in the rule `ClassWithOnlyPrivateConstructorsShouldBeFinal` as `@AllArgsConstructor` annotation was added to the filter that assumes the annotation always generates a public constructor at [ClassWithOnlyPrivateConstructorsShouldBeFinal.java](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/ClassWithOnlyPrivateConstructorsShouldBeFinalRule.java#L40). However, PMD does not consider the parameters of the annotation. When using the `staticName="build"` , the class does not generate a public constructor, leading to a false negative. Case2 is equivalent to Case1, and PMD reported a warning.

**Code Sample demonstrating the issue:**

**Case1**
```java
import lombok.AllArgsConstructor;
@AllArgsConstructor(staticName = "build")
public class Person {    //report no warnings
    private String name;
    private  int age;
    private Person(){}
}
```
**Case2**
```java
public class Person {
    private String name;
    private int age;

    private Person() {    //report a warning
    }

    @java.lang.SuppressWarnings("all")
    private Person(final String name, final int age) {
        this.name = name;
        this.age = age;
    }

    @java.lang.SuppressWarnings("all")
    public static Person build(final String name, final int age) {
        return new Person(name, age);
    }
}
```
**Expected outcome:**

PMD does not report a violation at line 3 in the first case, but that's wrong. That's a false negative.

**Running PMD through:** *[CLI]*

**Comment by Chiky777** [comment]
`@RequiredArgsConstructor(staticName="build")` and `@NoArgsConstructor(staticName="build")` will also cause this issue
